### DISCUSSION TITLE
Auto Map of complex objects

### DISCUSSION THREAD

**Discussion by SushmaReddyLoka** [atlasmap/atlasmap#4622] [Q&A]
Do you have any plans for supporting auto mapping for complex objects. ? 

I have tried to come up with the changes to support auto-map feature, and here is the draft PR for your reference https://github.com/atlasmap/atlasmap/pull/4621 . 
Can you please help in taking it forward by reviewing it.

@igarashitm @rhuss @chirino @christophd @WillemJiang

**Comment by ChandraSekharBk** [comment]
Hi @igarashitm , @rhuss , @chirino , @christophd , @WillemJiang ,

Could you please check this post and share your thoughts?

Also, please suggest on how to request or contribute to enhance AtlasMap implementation.

Thanks,
Chandra Sekhar B K

**Comment by rhuss** [comment]
Sorry, I've been disconnected from this project for many years, so I'm not of much help here.

**Comment by igarashitm** [comment]
I was the last active contributor for this project and I moved to the other project. At this moment, there is no active contributor for this project unfortunately.

**Reply by ChandraSekharBk** [reply]
Ok. Thanks for the reply @rhuss & @igarashitm. 

Can you suggest on how to onboard new contributors? Can we plan for sync-up to discuss about enabling others, so that this project can be moved active state?

Thanks,
Chandra

**Reply by igarashitm** [reply]
I'm afraid that's not likely to happen. However, since this is a public repo, you can fork and do whatever you want.

**Reply by mattrpav** [reply]
Hi @igarashitm - can you add me back as an administrator for the repo to do maint releases for the users?

**Reply by igarashitm** [reply]
@mattrpav Cool - added admin role to you.

**Comment by janstey** [comment]
@mattrpav that's a nice offer! I'm sure users would appreciate some up to date patches. Fair warning though that there's been very little activity in this project for a while now.

Also hope you're doing well. I still have fond memories of the Bentonville trip :-)

**Comment by SushmaReddyLoka** [comment]
@mattrpav , will you be able to review the above raised PR https://github.com/atlasmap/atlasmap/pull/4621 to support auto mapping of complex objects ?

**Comment by mattrpav** [comment]
Hi @SushmaReddyLoka- The PR looks to have a lot of changes, not just the auto-mapping of complex objects. To start, it would be best to break out things like README and pom file changes to separate PRs.

Realistically, it will be a bit to get the project modernized. I think it needs a fresh release based on dependency updates before starting to add new features.

Will you please draft up an issue ticket and detail the design goals and sample of how you would expect automapping of complex objects to behave?  

I can think of some problematic scenarios that need to be understood-- 

How to map a string to boolean field? 
  String vipCustomer = source.isVIPCustomer() // returns 'Yes'
  target.setVIPCustomer(boolean vipCustomer); // ?

How to map a string to number? 
 String price = source.getPrice() // returns $2.20
 target.setPrice(double price) // how to strip currency symbol?

**Comment by SushmaReddyLoka** [comment]
Sure @mattrpav , I will create a new ticket detailing out the expected behavior for auto mapping of complex/collection objects. 
Can you provide timelines by when the new release with dependency updates will be available.

**Comment by SushmaReddyLoka** [comment]
@mattrpav , can provide Estimated time by when we can expect new release ?

**Comment by mattrpav** [comment]
I don’t know when the release will be. I’m primarily focused on ActiveMQ.

**Reply by SushmaReddyLoka** [reply]
Hi @mattrpav , created issue of enhancement category https://github.com/atlasmap/atlasmap/issues/4627 . Please go through and let me know your thoughts, questions.
